tool/tas.re

```diff
@@ -127,6 +128,29 @@ static TAS_COMPONENT = Component {
                 Tas::set_velocity(frame.velocity);
                 Tas::set_acceleration(frame.acceleration);
                 TAS_STATE.replay_index += 1;
+            },
+            Replaying::PositionsAndInputs => {
+                let frame = TAS_STATE.recording.get(TAS_STATE.replay_index).unwrap();
+                Tas::set_location(frame.location);
+                Tas::set_rotation(frame.rotation);
+                Tas::set_velocity(frame.velocity);
+                Tas::set_acceleration(frame.acceleration);
```

Comment:
Refactor this out into a method on `TasState` and call it from here and `Replaying::Positions` and `Replaying::Inputs if replay_index == 0`.